src/admin/wordlift_admin_save_post.php

```diff
@@ -101,3 +101,150 @@ function wl_update_post_status( $post_id, $status ) {
+}
+add_action( 'admin_menu', 'wl_add_excerpt_meta_box' );
+ 
+/**
+ * Create fields in excerpt meta box.
+ *
+ * @param none
+ */
+function wl_add_field_excerpt_meta_boxes() {
+    global $post;
+    $post_id = $post->ID;
+    $wlSummariesExcerpt = get_post_meta($post_id, 'wl-summaries-excerpt',true);
+    ?>
+        <label class="screen-reader-text" for="excerpt"><?php echo esc_html__('Excerpt', 'wordlift');?></label>
+        <textarea rows="1" cols="40" name="excerpt" id="excerpt"><?php echo $post->post_excerpt;?></textarea>
+        <p><?php echo esc_html__( 'Excerpts are optional hand-crafted summaries of your content that can be used in your theme.', 'wordlift' );?><a href="https://codex.wordpress.org/Excerpt"><?php echo esc_html__('Learn more about manual excerpts.', 'wordlift');?></a></p>
+        <label class="screen-reader-text" for="excerpt_summary"><?php echo esc_html__('Excerpt', 'wordlift');?></label>
+        <textarea rows="1" cols="40" name="excerpt_summary" id="excerpt_summary"><?php if ($wlSummariesExcerpt) echo strip_tags($wlSummariesExcerpt);
+        else
+            echo ""; ?></textarea>
+
+        <p><?php echo esc_html__( 'WordLift generated excerpt.', 'wordlift' );?></p> 
+        <div class="wl-use-ref">
+            <button class="btn" id="wl-use-summary"><?php echo esc_html__('Use', 'wordlift');?></button>
+        
+            <button class="btn" id="wl-refresh-summary" data-post-id="<?php echo $post_id;?>"><?php echo esc_html__('Refresh', 'wordlift');?></button>
+       
+            <img id='loader' src="<?php echo plugin_dir_url(__FILE__).'../images/loading.gif';?>">
+        </div>
+        
+        
+    <?php
+}
+
+/**
+ * Refresh call to summarize API.
+ *
+ * @param int    $post_id The post ID
+ * @param string $status  The new status
+ */
+function wl_refresh_excerpt_summary() {
+    $post_id = intval( $_POST['post_id'] );
+    if ( isset( $post_id ) ) {
+        $rest = get_excerpt_summary($post_id);
+        $res = json_decode($rest,true); 
+        if ($res) {
+            if ( array_key_exists( 'summary', $res ) ) {
+                $summary = $res['summary'];
+                update_post_meta( $postid, "wl-summaries-excerpt", $summary );
+            }
+        }
+        $response = array(
+                    'status'=> 1,
+                );
+        echo json_encode($response);
+        die();
+    }
+}
+add_action( 'wp_ajax_wl_refresh_excerpt_summary', 'wl_refresh_excerpt_summary' );
+add_action( 'wp_ajax_nopriv_wl_refresh_excerpt_summary', 'wl_refresh_excerpt_summary' );
+
+/**
+ * Add backend script.
+ */
+function wl_admin_backend_script_style()
+{
+    global $post;
+    $wlSummariesExcerpt = get_post_meta($post->ID, 'wl-summaries-excerpt',true);
+
+    $jsvar = array(
+            'wpadminajax'   => admin_url('admin-ajax.php'),
+            'summaryKey'    => $wlSummariesExcerpt
+        );
+    wp_register_script('ets_admin_script', plugin_dir_url(__FILE__).'js/ets-admin.js', array());
```

Comment:
@ravisoni6262 JS should be transpiled.

Reply:
@ziodave Could you please share an example from the plugin itself, I will follow that.